### ISSUE TITLE
[BUG] Class `transition-transform` throws the wrong warning with the `tailwindcss/migration-from-tailwind-2` rule

### ISSUE THREAD

**Issue by Serator** [francoismassart/eslint-plugin-tailwindcss#77]
**Describe the bug**

When you insert the `transition-transform` class, you get an invalid warning:
```
Classname `transition-transform` is not needed in Tailwind CSS v3! | eslinttailwindcss/migration-from-tailwind-2
```
The problem is related to a RegExp in the "tailwindcss/migration-from-tailwind-2" rule.

Here are a few more examples:
* `transition-[var(--transform)]` - fail.
* `transition-[transform]` - correct.
* `[transform:scale(1)]` - correct.
* This problem also occurs with the `filter` and `backdrop-filter` classes.

**To Reproduce**
1. Type `transition-transform` in your code.
2. Run lint
3. You will get the message.

**Expected behavior**
No warning would happen.

**Environment (please complete the following information):**
 - OS: Windows 10
 - Softwares + version used:
     -  VSCode: 1.63.2
     - eslint-plugin-tailwindcss: 3.0.0
     - tailwindcss: 3.0.7

**Comment by francoismassart** [comment]
I believe `backdrop-filter` is deprecated...

**Comment by francoismassart** [comment]
FIXED in `3.0.1`

**Comment by Serator** [comment]
Yeah, `backdrop-filter` is obsolete, but it can be used as a value for `transition`.
